src/Integration/Telemetry/TelemetryManager.cs

```diff
@@ -154,9 +158,10 @@ public void Update()
             try
             {
                 var lastAnalysisDate = telemetryRepository.Data.LastSavedAnalysisDate;
-                if (!DateTimeOffset.Now.IsSameDay(lastAnalysisDate))
+                var now = currentTimeProvider.Now;
```

Comment:
This is technically a change of behaviour since we are now only calling `Now` once, but it's not material (and it we did `Now` twice it would make the tests slightly more complicated).